packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/ReactHostImpl.java

```diff
@@ -567,7 +569,23 @@ FabricUIManager getUIManager() {
     return new ArrayList<>();
   }
 
-  /* package */ @Nullable
+  void executeRuntimeExecutor(ReactRuntimeExecutor reactRuntimeExecutor) {
+    mReactInstanceTaskRef
+        .get()
+        .onSuccess(
+            task -> {
+              final ReactInstance reactInstance = task.getResult();
+              if (reactInstance == null) {
+                return false;
+              }
+              reactRuntimeExecutor.execute(reactInstance.getBufferedRuntimeExecutor());
+              return true;
+            },
+            mBGExecutor);
+  }
```

Comment:
Access to the runtime executer will be obtained asynchronously. Currently, in our flow, module initialization starts when native constants are received. It is uncertain whether this will be invoked in time before the JS continues executing.

Reply:
@lukmccall, the code is very different now.

NativeModules can now access the runtime by calling ReactContext.getRuntimeExecutor().

This method returns null when ReactHost's ReactInstance is/becomes null. 

This happens in two scenarios: 
1. The native module is created eagerly (i.e: during ReactInstance create/init).
2. The native module outlives the ReactInstance (i.e: ReactHost kills the ReactInstance, but something else keeps the native module alive).

The native module that initializes expomodules: is it eagerly initialized? If so, then we could have a problem.

Aside from that, do your concerns translate across Phillip's latest changes?